### PR TITLE
:sparkles: add dev_tools_ui in config

### LINKED ISSUE #674: Add enable dev tools config
this is a follow up task of [dash-renderer #100](https://github.com/plotly/dash-renderer/pull/100)

it adds an extra dev-tools config in `<script id="_dash-config" type="application/json">` section, the logic is simple, the value equals to the `debug` option in `run_server()`.

user set app.run_server(debug=True) will implicitly enable the dev-tools icon in FE.

### PR DESCRIPTION
**Pull request by byronz**
this fixes #674 

This PR introduces a new config `dev_tools_ui` to let FE know about if user has set debug=True. this implicitly enables the dev tools UI 

image

### CODE REVIEW

File: dash/dash.py [hunk 1]
```diff
@@ -328,7 +329,8 @@ def serve_layout(self):
     def _config(self):
         config = {
             'url_base_pathname': self.url_base_pathname,
-            'requests_pathname_prefix': self.config['requests_pathname_prefix']
+            'requests_pathname_prefix': self.config['requests_pathname_prefix'],  # noqa: E501
```
Comment (T4rk1n) [inline dash/dash.py]: Could also use the dot notation for less characters: `self.config.requests_pathname_prefix`.

Comment (alexcjohnson) [reply dash/dash.py]: Oh interesting, didn't realize we had a js-like dict class `AttributeDict`. FWIW that came in 18 months ago: https://github.com/plotly/dash/commit/885340256a92d2242c622a8e57fe7064890e80cb for backward compatibility with an old class-based config. Honestly I find that a bit confusing and un-pythonic, has it been long enough that we could get rid of the dot version?

Would also be nice to remove the spelling mistake fallback `supress_callback_exceptions` and just throw an error if someone tries to use the misspelled version. Maybe even list the allowed keys explicitly and throw a helpful error if any other key is provided.

Comment (byronz) [reply dash/dash.py]: I used this library `attrdict` before, and mostly the use case is for config.  yes we should create an issue for supress typo if there is no one yet
https://pypi.org/project/attrdict/

**Comment by nicolaskruchten** [comment]
Does this PR enable setting debug=True and dev_tools_ui=False, which would hide the UI?

**Comment by byronz** [comment]
> Does this PR enable setting debug=True and dev_tools_ui=False, which would hide the UI?

this PR is necessary if we wanna explicitly set the switch in the renderer

**Comment by nicolaskruchten** [comment]
I'm not sure you answered my question... Is it possible with this PR to set debug mode to True *except for* the dev-tools UI?

**Comment by byronz** [comment]
@nicolaskruchten there is no such `dev_tools_ui=False` in python code, it just adds a config field (redux state will parse the same config) which binds directly the value of `debug`.

**Pull request merged**